test/services/CreditHistoryServiceSpec.scala

```diff
@@ -77,15 +77,26 @@ class CreditHistoryServiceSpec extends TestSupport with MockIncomeTaxViewChangeC
       "return a list of MFA credits only" in {
         setupGetPayments(taxYear)(Payments(paymentsForTheGivenTaxYear))
         setupMockGetFinancialDetails(taxYear, nino)(taxYearFinancialDetails)
-        TestCreditHistoryService.getCreditsHistory(taxYear, nino).futureValue shouldBe Right(creditModels)
+        val mfaCreditModels = TestCreditHistoryService.getCreditsHistory(taxYear, nino).futureValue
+        mfaCreditModels shouldBe Right(creditModels)
+        mfaCreditModels.right.foreach { ds =>
+          ds.foreach{ d =>
+            d.creditType shouldBe MfaCreditType
+          }
+        }
       }
 
       "return a list of MFA and CutOver credits" in {
         setupGetPayments(taxYear)(Payments(creditsForTheGivenTaxYear))
         setupMockGetFinancialDetails(taxYear, nino)(taxYearFinancialDetails)
         setupGetPaymentAllocationCharges(nino, documentIdA)(cutOverCreditsAsFinancialDocumentA)
         setupGetPaymentAllocationCharges(nino, documentIdB)(cutOverCreditsAsFinancialDocumentB)
-        TestCreditHistoryService.getCreditsHistory(taxYear, nino).futureValue shouldBe Right(List(cutOverCreditA, cutOverCreditB) ++ creditModels)
+        val cs = TestCreditHistoryService.getCreditsHistory(taxYear, nino).futureValue
+        cs shouldBe Right(cutOverCreditA ++ cutOverCreditB ++ creditModels)
+        cs.right.foreach { ds =>
+          ds.filter(_.creditType == CutOverCreditType).length shouldBe (cutOverCreditA ++ cutOverCreditB).length
```

Comment:
Can be replaced with **count**: 
`          ds.count(_.creditType == CutOverCreditType) shouldBe (cutOverCreditA ++ cutOverCreditB).length
           ds.count(_.creditType == MfaCreditType) shouldBe creditModels.length
`